internal/configs/ingress.go

```diff
@@ -181,8 +181,14 @@ func generateNginxCfg(ingEx *IngressEx, pems map[string]string, apResources map[
 
 			ssl := sslServices[path.Backend.ServiceName] || staticParams.SpiffeCerts
 			proxySSLName := generateProxySSLName(path.Backend.ServiceName, ingEx.Ingress.Namespace)
+			var pathType extensions.PathType
+			if path.PathType == nil {
```

Comment:
I suggest placing all the path related logic into a singe function ```func generateIngressPath(path, pathType) string```, which takes a path and pathType (could be nil) and generate the appropriate path we can use in a location.  The function can have a unit test, similar to `TestGeneratePath` in  internal/configs/virtualserver_test.go:4481 

You can use this function in `createLocation`.

It will be similar to `generatePath` in internal/configs/virtualserver.go:752

Reply:
some of the logic will have to be before the `generateLocation` function is called because `Pathtype` is a pointer and passing that as a parameter ( if its nil ) will cause a `nil pointer dereference` error

Reply:
@LorcanMcVeigh 
> some of the logic will have to be before the generateLocation function is called because Pathtype is a pointer and passing that as a parameter ( if its nil ) will cause a nil pointer dereference error

it will not cause a a nil pointer dereference error use `*networking.PathType` as a parameter and if you check for nil in `generateIngressPath`:
```
func generateIngressPath(path string, pathType *networking.PathType) string {
	if pathType == nil {
           . . .
        }
       . . .
```

Or perhaps I'm missing something? 🤔